it/src/test/java/apoc/it/core/SchemasEnterpriseFeaturesTest.java

```diff
@@ -164,6 +164,93 @@ private static void schemaNodeKeyAssertions(Map<String, Object> r) {
     }
 
 
+    @Test
+    public void testSchemaNodesWithNodeTypeConstraint() {
+        session.writeTransaction(tx -> {
+            tx.run("CREATE CONSTRAINT node_prop_type_movie FOR (m:Movie) REQUIRE (m.first) IS :: INTEGER");
+            tx.commit();
+            return null;
+        });
+
+        testResult(session, CALL_SCHEMA_NODES_ORDERED, (result) -> {
+            Map<String, Object> r = result.next();
+            assertEquals("Movie", r.get("label"));
+            assertEquals(List.of("first"), r.get("properties"));
+            assertEquals(":Movie(first)", r.get("name"));
```

Comment:
This seems incorrect as name of the type constraint?

Reply:
I gave up questioning APOC :P It is consistent with how it works today (see docs here[https://neo4j.com/docs/apoc/current/overview/apoc.schema/apoc.schema.nodes/](url)). I agree it is weird, but I wouldn't change it here, would you like me to make a card for it?

Reply:
Let's make a card for it please